ui/components/app/connected-sites-list/connected-snaps.js

```diff
@@ -0,0 +1,131 @@
+
+export default function ConnectedSnaps({ connectedSubjects }) {
+  const [showOptions, setShowOptions] = useState();
+  const t = useI18nContext();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const subjects = useSelector((state) => getPermissionSubjects(state));
+  const connectedOrigin = useSelector(getOriginOfCurrentTab);
+
+  const onDisconnect = (snapId) => {
+    const caveatValue =
+      subjects[connectedOrigin].permissions[WALLET_SNAP_PERMISSION_KEY]
+        .caveats[0].value;
+    const newCaveatValue = { ...caveatValue };
+    delete newCaveatValue[snapId];
+    if (Object.keys(newCaveatValue).length > 0) {
+      dispatch(
+        updateCaveat(
+          connectedOrigin,
+          WALLET_SNAP_PERMISSION_KEY,
+          SnapCaveatType.SnapIds,
+          newCaveatValue,
+        ),
+      );
+    } else {
+      dispatch(
+        removePermissionsFor({
+          [connectedOrigin]: [WALLET_SNAP_PERMISSION_KEY],
+        }),
+      );
+    }
+  };
+
+  const renderListItemOptions = (snapId) => {
+    return (
+      <ConnectedAccountsListOptions
+        onHideOptions={() => setShowOptions()}
+        onShowOptions={() => setShowOptions(snapId)}
+        show={showOptions === snapId}
+      >
+        <MenuItem
+          iconName={IconName.Logout}
+          onClick={(e) => {
+            e.preventDefault();
+            onDisconnect(snapId);
+          }}
+        >
+          {t('disconnect')}
+        </MenuItem>
+        <MenuItem
+          iconName={IconName.Setting}
+          onClick={() =>
+            history.push(`${SNAPS_VIEW_ROUTE}/${encodeURIComponent(snapId)}`)
+          }
+        >
+          {t('snapsSettings')}
+        </MenuItem>
+      </ConnectedAccountsListOptions>
+    );
+  };
+
+  return (
+    <Box className="connected-snaps-list__content-rows" width={BlockSize.Full}>
+      {connectedSubjects.map((subject) => (
+        <Box
+          key={subject.origin}
+          className="connected-snaps-list__content-row"
+          width={BlockSize.Full}
+          display={Display.Flex}
+          flexDirection={FlexDirection.Row}
+          padding={4}
+          justifyContent={JustifyContent.spaceBetween}
+        >
+          <Box
+            className="connected-snaps-list__subject-info"
+            gap={4}
+            display={Display.Flex}
+            alignItems={AlignItems.center}
+          >
+            <SnapAvatar
+              snapId={subject.origin}
+              badgeSize={IconSize.Xs}
+              avatarSize={IconSize.Md}
+            />
+            <Text
+              variant={TextVariant.bodyLgMedium}
+              className="connected-accounts-list__account-name"
+            >
+              {subject.name}
+            </Text>
+          </Box>
+          {renderListItemOptions(subject.origin)}
+        </Box>
+      ))}
+    </Box>
+  );
+}
+
+ConnectedSnaps.propTypes = {
+  connectedSubjects: PropTypes.arrayOf(
```

Comment:
We should add documentation comments here as well.

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/20811/commits/2ba5304bb2d3f6c43ae05731a3ed4d15c83fd963